src/darker/command_line.py

```diff
@@ -127,4 +139,26 @@ def parse_command_line(argv: List[str]) -> Namespace:
         dest="line_length",
         help="How many characters per line to allow [default: 88]",
     )
-    return parser.parse_args(argv)
+    return parser
+
+
+def parse_command_line(argv: List[str]) -> Tuple[Namespace, DarkerConfig, DarkerConfig]:
+    """Return the parsed command line, using defaults from a configuration file
+
+    Also return the effective configuration which combines defaults, the configuration
+    read from ``pyproject.toml`` (or the path given in ``--config``), and command line
+    arguments.
+
+    Finally, also return the set of configuration options which differ from defaults.
+
+    """
+    config = load_config()
+    parser = make_argument_parser(require_src=not config.get("src"))
```

Comment:
I think that should always be required, and that load_config sound take that as a parameter to find the project root.

I think mainly in editors the CWD may not be "in" the project; so relying on global state is a code smell for me.

Reply:
What I'm aiming at here is a convenience configuration: if I specify e.g. the main code directory using
```toml
[tool.darker]
src = [
    "src",
]
```
in `pyproject.toml`, then running a plain `darker` command without arguments in that directory should do the right thing and check files in the `src/` tree.

This is somewhat analogous to e.g. `pytest` – according to [documentation](https://docs.pytest.org/en/stable/customize.html#pyproject-toml), in `pyproject.toml` you can write:
```toml
[tool.pytest.ini_options]
testpaths = [
    "tests",
    "integration",
]
```
and running just `pytest` will discover tests only in `tests/` and `integration/`.

What I did here work so that if you _don't_ specify `src = [...]` in `pyproject.toml` for Darker, it _will_ require at least one path on the command line and fail if none are provided.

Reply:
Ah, ok, I misunderstood. Yes, if running from the root of a project, doing `darker` without argument should probably work.

Reply:
...and let's say I have this:
```
.
├── pyproject.toml
└── src
    └── darker
        ├── __init__.py
        └── __main__.py
```
and this in `pyproject.toml`:
```toml
[tool.darker]
src = [
    "src",
]
```
Then should running `darker .` at the root be equivalent to
- `darker . src`,
- or just `darker src`?

Without `src = [...]` in `pyproject.toml` it's obviously equivalent just to `darker .`.

But anyway, I'll fix it so that `pyproject.toml` is searched for relative to any given paths on the command line. The behavior when no paths are given is debatable, let's iron that out.